src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -515,19 +515,25 @@ protected function deleteCollection(Document $database, Document $collection, Do
         $databaseId = $database->getId();
         $databaseInternalId = $database->getInternalId();
 
-        $relationships = \array_filter(
-            $collection->getAttribute('attributes'),
-            fn ($attribute) => $attribute['type'] === Database::VAR_RELATIONSHIP
-        );
+        /**
+         * Related collections relating to current collection
+         */
+        $attributes = $dbForProject->find('attributes', [
+            Query::equal('databaseInternalId', [$databaseInternalId]),
+            Query::equal('type', [Database::VAR_RELATIONSHIP]),
+            Query::notEqual('collectionInternalId', $collectionInternalId),
+            Query::contains('options', ['"relatedCollection":"'. $collectionId .'"']), // Comment in Version < 1.5
+            Query::limit(PHP_INT_MAX)
+        ]);
 
-        foreach ($relationships as $relationship) {
-            if (!$relationship['twoWay']) {
-                continue;
+        foreach ($attributes as $attribute) {
+            $options = $attribute->getAttribute('options', []);
+            if($options['relatedCollection'] === $collectionId) { // Remove if using contains query above
```

Comment:
I'm surprised our linter didn't catch this 🧐 
```suggestion
            if ($options['relatedCollection'] === $collectionId) { // Remove if using contains query above
```

Reply:
Since we are going to use 1.5 we do not need this if case , the contains query will bring the right  collection 👍